### PR TITLE
feat: edge mode dashboard service

### PR DESCRIPTION
**Pull request by chejimmy**
# Description

Added an local storage dashboard service for edge mode usage.

# How Has This Been Tested?

Unit tests added

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: apps/client/src/services/index.ts [hunk 1]
```diff
@@ -26,42 +54,30 @@ export function setServiceUrl(url: string) {
...
+  : (id) => DashboardsService.dashboardsControllerRead(id);
+export const updateDashboard: UpdateDashboard = isEdgeMode()
+  ? (id, dto) => updateDashboardLocal(id, dto)
+  : (id, dto) => DashboardsService.dashboardsControllerUpdate(id, dto);
+export const deleteDashboard: DeleteDashboard = isEdgeMode()
+  ? (id) => deleteDashboardLocal(id)
+  : (id) => DashboardsService.dashboardsControllerDelete(id);
+export const bulkDeleteDashboards: BulkDeleteDashboards = isEdgeMode()
+  ? (dto) => bulkDeleteDashboardsLocal(dto)
+  : (dto) => DashboardsService.dashboardsControllerBulkDelete(dto);
```
Comment (tracy-french) [inline apps/client/src/services/index.ts:74]: I think you'll want an additional set of integration tests to ensure `core` is used in the cloud and `localStorage` is used at the edge. Would it make sense to add these now or as a follow-up?

File: apps/client/iot-app-kit-dashboard.tgz
[the remark this answers is not in the document]
Comment (chejimmy) [reply apps/client/iot-app-kit-dashboard.tgz:1]: ops it shouldn't be there; deleting

File: apps/client/src/services/index.ts (hunk 1 above)
Comment (tracy-french) [reply apps/client/src/services/index.ts:74]: Sounds good!

**Review by tracy-french — APPROVED** [review]
Great work! Thank you @chejimmy!

File: apps/client/src/services/local-storage-dashboard/dashboard-service.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+};
+
+const getLocalStorageDashboard = (id: string) => {
+  const dashboards = getLocalStorageDashboards();
+
+  const dashboard = dashboards.find((dashboard) => dashboard.id === id);
+  return dashboard;
+};
+
+if (import.meta.vitest) {
```
Comment (dpportet) [inline apps/client/src/services/local-storage-dashboard/dashboard-service.ts:154]: Is there a reason to now have a separate test file?

Comment (chejimmy) [reply apps/client/src/services/local-storage-dashboard/dashboard-service.ts:154]: nothing particular; just to simplify the module so it doesn't need to export things that has no value outside (e.g. `LOCAL_STORAGE_DASHBOARDS_NAMESPACE`)

**Pull request merged**